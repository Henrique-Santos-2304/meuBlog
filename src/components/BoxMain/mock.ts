export const mockTypes = {
  data: {
    name: "Henrique dos Santos",
    profission: "Desenvolvedor FrontEnd",
    platform: "Web, Mobile, Desktop",
    linksSocials: [],
    email: {
      title: "icons",
      email: "henrique.multitech@gmail.com",
      ariaLabel: "Envio de email via site",
      id: "name",
    },
  },
};
