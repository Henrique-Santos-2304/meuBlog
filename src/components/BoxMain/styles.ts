import styled from "styled-components";

export const Span = styled.span`
  width: 100%;
  position: relative;
  margin-bottom: 2rem;
  :after {
    position: absolute;
    top: 0;
    left: 50%;
    transform: translateX(-50%);
    content: "";
    width: 30%;
    height: 0.6rem;
    background: ${({ theme }) => theme.colors.yellow};
    border-radius: 2rem;
  }
`;
