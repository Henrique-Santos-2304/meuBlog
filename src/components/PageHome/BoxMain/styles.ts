import styled, { css } from "styled-components";
export const Wrapper = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-top: -7rem;
  margin-bottom: 2rem;
`;
export const Name = styled.h1`
  ${({ theme }) => css`
    font-size: calc(${theme.font.sizes.xxbiglarge} + 1rem);
    font-family: ${theme.font.family.Tangerine};
    text-align: center;
    color: ${theme.colors.textTitles};

    @media (min-width: ${theme.media.medium}) {
      font-size: ${theme.font.sizes.ultra};
    }
  `}
`;
export const Profission = styled.h3`
  font-size: ${({ theme }) => theme.font.sizes.large};
  color: ${({ theme }) => theme.colors.mainLight};
  margin-top: -1.3rem;
`;
export const Platforms = styled.p`
  ${({ theme }) => css`
    text-align: center;
    font-family: ${theme.font.family.Poppins};
    font-size: ${theme.font.sizes.small};
    font-style: italic;
    color: ${({ theme }) => theme.colors.mainLight};
    margin: 0.8rem 0 2rem;
  `}
`;
export const EffectTopText = styled.span`
  width: 100%;
  position: relative;
  margin-bottom: 0.4rem;
  :after {
    position: absolute;
    top: 0;
    left: 50%;
    transform: translateX(-50%);
    content: "";
    width: 30%;
    height: 0.6rem;
    background: ${({ theme }) => theme.colors.textTitles};
    border-radius: 2rem;
  }
`;
