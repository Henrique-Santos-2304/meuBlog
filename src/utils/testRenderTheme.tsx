import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
