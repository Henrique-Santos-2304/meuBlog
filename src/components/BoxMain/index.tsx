import { Flex, Heading, Text, useMediaQuery } from "@chakra-ui/react";
import IconsContact from "components/IconsContact";
import theme from "styles/theme";
import * as S from "./styles";

const BoxMain = () => {
  const [isLargerThan768] = useMediaQuery("(min-width: 768px)");
  return (
    <Flex
      as="main"
      align="center"
      direction="column"
      mt={["1rem", null, "-5rem"]}
      mb="3rem"
      aria-label="Descrições de um Desenvolvedor FrontEnd"
    >
      <S.Span></S.Span>
      <Heading
        as="h1"
        fontSize={["3.2rem", null, "3.4rem", "4.3rem"]}
        letterSpacing="0.1rem"
        isTruncated
        p="0.7rem"
        color="red.600"
        fontFamily={theme.font.family.GrechenFuemen}
      >
        Henrique dos Santos
      </Heading>
      <Heading
        as="h4"
        fontSize={["1.7rem", null, "1.9rem"]}
        isTruncated
        m="0.5rem 1rem"
        color="gray.100"
        fontStyle="italic"
        fontFamily={theme.font.family.Poppins}
      >
        Desenvolvedor FrontEnd
      </Heading>
      <Text
        lineHeight="4rem"
        color="gray.100"
        fontStyle="italic"
        fontFamily={theme.font.family.Poppins}
      >
        Mobile, Web, Desktop
      </Text>
      {isLargerThan768 && <IconsContact />}
    </Flex>
  );
};

export default BoxMain;
