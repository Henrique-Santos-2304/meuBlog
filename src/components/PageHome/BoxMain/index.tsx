import * as S from "./styles";
import HandleMediaQuery from "components/globalComponents/MediaQuery";

import { queryHome_home_boxMain } from "graphql/typesFinal/queryHome";
import IconsContact from "components/PageHome/IconsContact";

type props = {
  data: queryHome_home_boxMain;
};
const BoxMain = ({ data }: props) => {
  const iconLink = {
    linksSocials: data.linksSocials,
    email: data.email,
  };
  return (
    <S.Wrapper aria-Label="Descrições de um Desenvolvedor FrontEnd">
      <S.EffectTopText></S.EffectTopText>
      <S.Name>{data.name}</S.Name>
      <S.Profission>{data.profission}</S.Profission>
      <S.Platforms>{data.platform}</S.Platforms>

      <HandleMediaQuery responsive="min-width" asWidth="767px" show={true}>
        <IconsContact contacts={iconLink} />
      </HandleMediaQuery>
    </S.Wrapper>
  );
};

export default BoxMain;
