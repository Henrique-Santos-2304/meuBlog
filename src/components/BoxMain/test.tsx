/* eslint-disable @typescript-eslint/no-empty-function */
import BoxMain from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

jest.mock("@chakra-ui/react", () => {
  const originalModule = jest.requireActual("@chakra-ui/react");

  return {
    __esModule: true,
    ...originalModule,
    useMediaQuery: jest.fn().mockImplementation(() => [true]),
  };
});
describe("<BoxMain />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<BoxMain />);
    expect(container).toMatchSnapshot();
  });
  it("should render ", () => {
    renderTheme(<BoxMain />);
    expect(
      screen.getByRole("heading", {
        name: /henrique dos santos/i,
      })
    ).toBeInTheDocument();
    expect(
      screen.getByRole("heading", {
        name: /Desenvolvedor FrontEnd/i,
      })
    ).toBeInTheDocument();
  });

  it("should have as link for my Contacts and portfÃ³lio ", () => {
    renderTheme(<BoxMain />);
    expect(screen.getByLabelText(/Link Para meu Github/i)).toBeInTheDocument();
    expect(
      screen.queryByLabelText(/Link Para meu Linkedin/i)
    ).toBeInTheDocument();
    expect(
      screen.queryByLabelText(/Envio de email via site/i)
    ).toBeInTheDocument();
    expect(
      screen.queryByLabelText(/Envio de mensagem no Whatsapp/i)
    ).toBeInTheDocument();
  });
});
