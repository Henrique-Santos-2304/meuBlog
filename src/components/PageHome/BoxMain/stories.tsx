import { Story, Meta } from "@storybook/react";
import BoxMain from ".";

export default {
  title: "BoxMain",
  component: BoxMain,
} as Meta;

export const Basic: Story = (args) => <BoxMain {...args} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <BoxMain {...args} />;
