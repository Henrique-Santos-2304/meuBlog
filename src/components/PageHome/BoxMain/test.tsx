/* eslint-disable @typescript-eslint/no-empty-function */
import BoxMain from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";
import { mockTypes } from "./mockBoxMain";

jest.mock("@chakra-ui/react", () => {
  const originalModule = jest.requireActual("@chakra-ui/react");

  return {
    __esModule: true,
    ...originalModule,
    useMediaQuery: jest.fn().mockImplementation(() => [true]),
  };
});

describe("<BoxMain />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<BoxMain {...mockTypes} />);
    expect(container).toMatchSnapshot();
  });
  it("should render ", () => {
    renderTheme(<BoxMain {...mockTypes} />);
    expect(
      screen.getByRole("heading", {
        name: /henrique dos santos/i,
      })
    ).toBeInTheDocument();
    expect(
      screen.getByRole("heading", {
        name: /Desenvolvedor FrontEnd/i,
      })
    ).toBeInTheDocument();
  });
});
