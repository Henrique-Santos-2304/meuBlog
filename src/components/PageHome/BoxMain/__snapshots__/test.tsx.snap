// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoxMain /> should render  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: -7rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: calc(3.4rem + 1rem);
  font-family: 'Tangerine',cursive;
  text-align: center;
  color: #F4F66F;
}

.c3 {
  font-size: 1.8rem;
  color: #f8f8f8;
  margin-top: -1.3rem;
}

.c4 {
  text-align: center;
  font-family: 'Poppins',sans-serif;
  font-size: 1.4rem;
  font-style: italic;
  color: #f8f8f8;
  margin: 0.8rem 0 2rem;
}

.c1 {
  width: 100%;
  position: relative;
  margin-bottom: 0.4rem;
}

.c1:after {
  position: absolute;
  top: 0;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  content: "";
  width: 30%;
  height: 0.6rem;
  background: #F4F66F;
  border-radius: 2rem;
}

.c5 {
  display: none;
}

.c6 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c7 {
  border-radius: 50%;
  width: 3rem;
  height: 3rem;
  margin: -1rem 0.3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  -webkit-transition: all 0.4s ease-in;
  transition: all 0.4s ease-in;
  position: relative;
  cursor: pointer;
}

.c7:hover {
  -webkit-transform: scale(1.5);
  -ms-transform: scale(1.5);
  transform: scale(1.5);
  margin: 0 1rem;
}

.c7:hover svg {
  font-size: 1.8rem;
}

.c7:hover:hover::before {
  -webkit-transform: translate(0);
  -ms-transform: translate(0);
  transform: translate(0);
  -webkit-animation: hRyZbK 0.4s ease-in;
  animation: hRyZbK 0.4s ease-in;
}

.c7:hover::before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: linear-gradient(to top right,#a2afb3 20%,#47494f,#111 70%);
  z-index: -1;
  -webkit-transition: all 0.4s ease-in;
  transition: all 0.4s ease-in;
  -webkit-transform: translate(50%,50%);
  -ms-transform: translate(50%,50%);
  transform: translate(50%,50%);
}

.c8 {
  font-size: 2.4rem;
  border-radius: 50%;
  color: #F4F66F;
}

@media (min-width:767px) {
  .c2 {
    font-size: 5rem;
  }
}

@media (min-width:767px) {
  .c5 {
    display: block;
  }
}

<div>
   
  <main
    aria-label="Descrições de um Desenvolvedor FrontEnd"
    class="c0"
  >
    <span
      class="c1"
    />
    <h1
      class="c2"
    >
      Henrique dos Santos
    </h1>
    <h3
      class="c3"
    >
      Desenvolvedor FrontEnd
    </h3>
    <p
      class="c4"
    >
      Web, Mobile, Desktop
    </p>
    <div
      class="c5"
      data-testid="media-query"
    >
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <svg
            aria-label="Envio de email via site"
            class="c8"
            fill="currentColor"
            height="1em"
            id="icons"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              henrique.multitech@gmail.com
            </title>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
        </div>
      </div>
    </div>
  </main>
</div>
`;
